packages/shared_preferences/shared_preferences_android/lib/shared_preferences_android.dart

```diff
@@ -74,6 +75,7 @@ class SharedPreferencesAndroid extends SharedPreferencesStorePlatform {
         return _api.setStringList(key, value as List<String>);
     }
 
+    // TODO(tarrinneal): change to ArgumentError to match other implementations.
```

Comment:
Sorry, my comment was not clearly phrased. It's not that this should be an `ArgumentError` to match other implementations; they aren't using it either. It's that we should change this to be an `ArgumentError` simultaneously across all implementations, because that's a better way to express this failure case.